SQL/Archive/2018-05-28_publication_schema.sql

```diff
@@ -0,0 +1,169 @@
+  `Name` varchar(255) NOT NULL,
+  `Email` varchar(255),
+  PRIMARY KEY(`PublicationCollaboratorID`),
+  CONSTRAINT `UK_publication_collaborator_Email` UNIQUE (`Email`)
+) ENGINE=InnoDB DEFAULT CHARSET='utf8mb4';
+
+DROP TABLE IF EXISTS publication_collaborator_rel;
+CREATE TABLE `publication_collaborator_rel` (
+  `PublicationID` int(10) unsigned NOT NULL,
+  `PublicationCollaboratorID` int(10) unsigned NOT NULL,
+  PRIMARY KEY(PublicationID, PublicationCollaboratorID),
+  CONSTRAINT `FK_publication_collaborator_1` FOREIGN KEY(`PublicationID`) REFERENCES `publication` (`PublicationID`),
+  CONSTRAINT `FK_publication_collaborator_2` FOREIGN KEY(`PublicationCollaboratorID`) REFERENCES `publication_collaborator` (`PublicationCollaboratorID`)
+) ENGINE=InnoDB DEFAULT CHARSET='utf8mb4';
+
+-- Publication - Variable of Interest  relational table
+DROP TABLE IF EXISTS publication_parameter_type_rel;
+CREATE TABLE `publication_parameter_type_rel` (
+    `PublicationID` int(10) unsigned NOT NULL,
+    `ParameterTypeID` int(10) unsigned NOT NULL,
+    PRIMARY KEY (PublicationID, ParameterTypeID),
+    CONSTRAINT `FK_publication_parameter_type_rel_1` FOREIGN KEY (`PublicationID`) REFERENCES `publication` (`PublicationID`),
+    CONSTRAINT `FK_publication_parameter_type_rel_2` FOREIGN KEY (`ParameterTypeID`) REFERENCES `parameter_type` (`ParameterTypeID`)
+) ENGINE=InnoDB DEFAULT CHARSET='utf8mb4';
+
+DROP TABLE IF EXISTS publication_test_names_rel;
+CREATE TABLE `publication_test_names_rel` (
+    `PublicationID` int(10) unsigned NOT NULL,
+    `TestNameID` int(10) unsigned NOT NULL,
+    PRIMARY KEY(`PublicationID`, `TestNameID`),
+    CONSTRAINT `FK_publication_test_names_rel_1` FOREIGN KEY (`PublicationID`) REFERENCES `publication` (`PublicationID`),
+    CONSTRAINT `FK_publication_test_names_rel_2` FOREIGN KEY (`TestNameID`) REFERENCES `test_names` (`ID`)
+) ENGINE=InnoDB DEFAULT CHARSET='utf8mb4';
+
+-- Publication Uploads
+DROP TABLE IF EXISTS publication_upload_type;
+CREATE TABLE `publication_upload_type` (
+  `PublicationUploadTypeID` int(2) unsigned NOT NULL AUTO_INCREMENT,
+  `Label` varchar(255) NOT NULL,
+  PRIMARY KEY (`PublicationUploadTypeID`)
+) ENGINE=InnoDB DEFAULT CHARSET='utf8mb4';
+
+INSERT INTO publication_upload_type (`Label`) VALUES ('Paper');
+INSERT INTO publication_upload_type (`Label`) VALUES ('Poster');
+INSERT INTO publication_upload_type (`Label`) VALUES ('Presentation');
+INSERT INTO publication_upload_type (`Label`) VALUES ('Other');
+
+DROP TABLE IF EXISTS publication_upload;
+CREATE TABLE `publication_upload` (
+    `PublicationUploadID` int(10) unsigned NOT NULL AUTO_INCREMENT,
+    `PublicationID` int(10) unsigned NOT NULL,
+    `PublicationUploadTypeID` int(2) unsigned NOT NULL,
+    `URL` varchar(255) NOT NULL,
+    `Version` varchar(255),
+    `Citation` text,
+    PRIMARY KEY (`PublicationUploadID`),
+    CONSTRAINT `UK_publication_upload_1` UNIQUE (URL),
```

Comment:
Do multiple version of a publication have different URL or any version can use the same URL?

Reply:
multiple versions of a publication can have different URLs, given that they're given different file names. alternatively, users can upload a different version of the same publication with the same file name if they delete the older version.

Reply:
Ok thanks for the precision